BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2086,14 +2088,33 @@ IEnumerator MissileRoutine()
             if (animStates != null) StartCoroutine(FlightAnimRoutine());
             yield return new WaitForSecondsFixed(cruiseDelay);
             if (cruiseRangeTrigger > 0)
-                yield return new WaitUntilFixed(checkCruiseRangeTrigger);
+                yield return new WaitUntilFixed(this.checkCruiseRangeTrigger);
                 
             yield return StartCoroutine(CruiseRoutine());
         }
 
         bool checkCruiseRangeTrigger()
         {
-            return ((TargetPosition - vessel.CoM).sqrMagnitude < cruiseRangeTrigger * cruiseRangeTrigger);
+            float sqrRange = (this.TargetPosition - this.part.rb.position).sqrMagnitude;
+
+            if (BDArmorySettings.DEBUG_MISSILES) Debug.Log($"[BDArmory.MissileLauncher]: Check cruise range trigger range: {Mathf.Sqrt(sqrRange)}");
```

Comment:
`Mathf.Sqrt` can have issues on M1 macs in some situations, so it's better to use `BDAMath.Sqrt` instead.